ksp_plugin_adapter/flight_planner.cs

```diff
@@ -49,175 +54,194 @@ class FlightPlanner : SupervisedWindowRenderer {
-                    plugin_.FlightPlanGetDesiredFinalTime(vessel_guid);
-                burn_editors_.Add(new BurnEditor(adapter_,
-                                                 plugin_,
-                                                 vessel_,
-                                                 initial_time : 0));
-                burn_editors_.Last().Reset(
-                    plugin_.FlightPlanGetManoeuvre(vessel_guid, i));
-              }
-            } else {
-              if (UnityEngine.GUILayout.Button("Create flight plan")) {
-                plugin_.FlightPlanCreate(vessel_guid,
-                                         plugin_.CurrentTime() + 1000,
-                                         vessel_.GetTotalMass());
-                final_time_.value =
-                    plugin_.FlightPlanGetDesiredFinalTime(vessel_guid);
-                Shrink();
-              }
-            }
-          }
-        } else {
-          if (final_time_.Render(enabled: true)) {
-            plugin_.FlightPlanSetDesiredFinalTime(vessel_guid,
-                                                  final_time_.value);
+      if (burn_editors_ == null) {
+        string vessel_guid = vessel_?.id.ToString();
+        if (vessel_guid != null &&
+            plugin_.HasVessel(vessel_guid) &&
+            plugin_.FlightPlanExists(vessel_guid)) {
+          burn_editors_ = new List<BurnEditor>();
+          for (int i = 0;
+               i < plugin_.FlightPlanNumberOfManoeuvres(vessel_guid);
+               ++i) {
+            // Dummy initial time, we call |Reset| immediately afterwards.
             final_time_.value =
                 plugin_.FlightPlanGetDesiredFinalTime(vessel_guid);
+            burn_editors_.Add(new BurnEditor(adapter_,
+                                             plugin_,
+                                             vessel_,
+                                             initial_time : 0));
+            burn_editors_.Last().Reset(
+                plugin_.FlightPlanGetManoeuvre(vessel_guid, i));
           }
-          double actual_final_time =
-              plugin_.FlightPlanGetActualFinalTime(vessel_guid);
-          UnityEngine.GUILayout.TextField(
-              (final_time_.value == actual_final_time)
-                  ? ""
-                  : "Timed out after " +
-                        FormatPositiveTimeSpan(TimeSpan.FromSeconds(
-                            actual_final_time -
-                            plugin_.FlightPlanGetInitialTime(vessel_guid))));
+        }
+      }
+    }
+
+    // The UI code proper, executed identically for Layout and Repaint.
+    {
+      string vessel_guid = vessel_?.id.ToString();
+      if (vessel_guid == null ||
+          !plugin_.HasVessel(vessel_guid)) {
+        return;
+      }
 
-          FlightPlanAdaptiveStepParameters parameters =
-              plugin_.FlightPlanGetAdaptiveStepParameters(vessel_guid);
+      if (plugin_.FlightPlanExists(vessel_guid)) {
```

Comment:
Does this not change the set of controls between calls to OnGUI?
the `else if` branch below may change the value of this condition.

Reply:
Added a comment for the uninitiated.